store/mongo/datastore_mongo.go

```diff
@@ -366,7 +362,7 @@ func (db *DataStoreMongo) DeleteTokenByDevId(
 ) error {
 	c := db.client.Database(ctxstore.DbFromContext(ctx, DbName)).
 		Collection(DbTokensColl)
-	ci, err := c.DeleteMany(ctx, bson.M{"sub": devID})
+	ci, err := c.DeleteMany(ctx, ctxstore.WithTenantID(ctx, bson.M{"sub": devID}))
```

Comment:
```suggestion
	var tenantID string
	if id := identity.FromContext(ctx); id != nil {
		tenantID = id.Tenant
	}
	ci, err := c.DeleteMany(ctx, bson.M{"sub": devID, "mender.tenant": tenantID})
```

Reply:
By the way, there's no index backing this query. We should analyze the usage of this function (NOTE to self).